tests/components/homekit/test_accessories.py

```diff
@@ -155,6 +155,89 @@ def demo_func(*args):
     assert acc._char_low_battery.value == 0
     assert acc._char_charging.value == 0
 
+async def test_linked_battery_sensor(hass, hk_driver, caplog):
+    """Test battery service with linked_battery_sensor."""
+    entity_id = 'homekit.accessory'
+    linked_battery_entity_id = 'sensor.battery'
+    hass.states.async_set(linked_battery_entity_id, 50, None)
+    await hass.async_block_till_done()
+
+    acc = HomeAccessory(hass, hk_driver, 'Battery Service', entity_id, 2,
+                        {CONF_LINKED_BATTERY_SENSOR: linked_battery_entity_id})
+    acc.update_state = lambda x: None
+    assert acc._char_battery.value == 0
+    assert acc._char_low_battery.value == 0
+    assert acc._char_charging.value == 2
+
+    await hass.async_add_job(acc.run)
+    await hass.async_block_till_done()
+    assert acc._char_battery.value == 50
+    assert acc._char_low_battery.value == 0
+    assert acc._char_charging.value == 2
+
+    hass.states.async_set(linked_battery_entity_id, 15, None)
+    await hass.async_block_till_done()
+    assert acc._char_battery.value == 15
+    assert acc._char_low_battery.value == 1
+    assert acc._char_charging.value == 2
+
+    hass.states.async_set(linked_battery_entity_id, 'error', None)
+    await hass.async_block_till_done()
+    assert acc._char_battery.value == 15
+    assert acc._char_low_battery.value == 1
+    assert acc._char_charging.value == 2
+    assert 'ERROR' not in caplog.text
+
+    # Test charging
+    hass.states.async_set(linked_battery_entity_id, 10, {
+        ATTR_BATTERY_CHARGING: True})
+    await hass.async_block_till_done()
+
+    acc = HomeAccessory(hass, hk_driver, 'Battery Service', entity_id, 2,
+                        {CONF_LINKED_BATTERY_SENSOR: linked_battery_entity_id})
+    acc.update_state = lambda x: None
+    assert acc._char_battery.value == 0
+    assert acc._char_low_battery.value == 0
+    assert acc._char_charging.value == 2
+
+    await hass.async_add_job(acc.run)
+    await hass.async_block_till_done()
+    assert acc._char_battery.value == 10
+    assert acc._char_low_battery.value == 1
+    assert acc._char_charging.value == 1
+
+    hass.states.async_set(linked_battery_entity_id, 100, {
+        ATTR_BATTERY_CHARGING: False})
+    await hass.async_block_till_done()
+    assert acc._char_battery.value == 100
+    assert acc._char_low_battery.value == 0
+    assert acc._char_charging.value == 0
+
+    # Test override with linked battery sensor and existing battery attribute
```

Comment:
Good catch to include this. It would probably execute even faster if you integrate it with your first and second test cases.